### DISCUSSION TITLE
Accessing pydantic's parsing

### DISCUSSION THREAD

**Discussion by danhje** [pydantic/pydantic#7777] [General]
In v1 you could run `parse_datetime` manually. Is that still possible?

Use case:
Let's say you want to leverage pydantic's great coercion logic to turn the input value into a datetime, whether it's a string, int or datetime object. But you want the final value to be a float timestamp.

The below solution works, but gives the warning `UserWarning: Pydantic serializer warnings: Expected 'datetime' but got 'float' - serialized value may not be as expected`

```python
from datetime import datetime
from typing import Annotated

from pydantic import AfterValidator, BaseModel, TypeAdapter


FloatTimestamp = Annotated[datetime, AfterValidator(datetime.timestamp)]


class Model(BaseModel):
    ts: FloatTimestamp 


m = Model(ts='2023-10-09T15:29:52.655319')
print(m.model_dump())
# {'ts': 1696858192.655319}
```

The correct thing to do is probably to pass `float` as the first argument to `Annotated` as that is the final type, and use a `BeforeValidator` to do the initial coercion, but I don't want to implement it myself when pydantic already has that capability. I guess I could use a `TypeAdapter` in that `BeforeValidator`, but that seems a bit hairy.

**Comment by TonyMistark** [comment]
### There are warnings, feel less than perfect.
```python
/home/ice/.virtualenvs/311/lib/python3.11/site-packages/pydantic/main.py:308: UserWarning: Pydantic serializer warnings:
  Expected `datetime` but got `float` - serialized value may not be as expected
  return self.__pydantic_serializer__.to_python(
Out[37]: {'ts': 1696865392.655319}

```
